tests/test_compile_kwargs.py

```diff
@@ -0,0 +1,555 @@
+)
+@pytest.mark.parametrize("n_outputs_", (1, 2))
+def test_loss_routed_params_iterable(loss, n_outputs_):
+    """Tests compiling of loss when it is
+    given as an iterable of losses
+    mapping to outputs.
+    """
+
+    X, y = make_classification()
+    y = np.column_stack([y for _ in range(n_outputs_)]).squeeze()
+
+    # Test iterable with global routed param
+    est = KerasClassifier(
+        model=get_model,
+        num_hidden=20,
+        optimizer="sgd",
+        loss=[loss],
+        loss__from_logits=True,  # default is False
+    )
+    est.fit(X, y)
+    assert est.model_.loss[0].from_logits
+
+    # Test iterable with index-based routed param
+    est = KerasClassifier(
+        model=get_model,
+        num_hidden=20,
+        optimizer="sgd",
+        loss=[loss],
+        loss__from_logits=True,
+        loss__0__from_logits=False,  # should override above
+    )
+    est.fit(X, y)
+    assert est.model_.loss[0].from_logits == False
+
+
+@pytest.mark.parametrize(
+    "loss", (losses_module.BinaryCrossentropy, "BinaryCrossentropy",),
+)
+@pytest.mark.parametrize("n_outputs_", (1, 2))
+def test_loss_routed_params_dict(loss, n_outputs_):
+    """Tests compiling of loss when it is
+    given as an dict of losses
+    mapping to outputs.
+    """
+
+    X, y = make_classification()
+    y = np.column_stack([y for _ in range(n_outputs_)]).squeeze()
+
+    # Test dict with global routed param
+    est = KerasClassifier(
+        model=get_model,
+        num_hidden=20,
+        optimizer="sgd",
+        loss={"out1": loss},
+        loss__from_logits=True,  # default is False
+    )
+    est.fit(X, y)
+    assert est.model_.loss["out1"].from_logits == True
+
+    # Test dict with key-based routed param
+    est = KerasClassifier(
+        model=get_model,
+        num_hidden=20,
+        optimizer="sgd",
+        loss={"out1": loss},
+        loss__from_logits=True,
+        loss__out1__from_logits=False,  # should override above
+    )
+    est.fit(X, y)
+    assert est.model_.loss["out1"].from_logits == False
+
+
+@pytest.mark.parametrize(
+    "metrics", ["binary_accuracy", metrics_module.BinaryAccuracy]
+)
+@pytest.mark.parametrize("n_outputs_", (1, 2))
+def test_metrics_single_metric_per_output(metrics, n_outputs_):
+    """Metrics without the ("name", metric, "output")
+    syntax should ignore all routed and custom options.
+
+    This tests a single metric per output.
+    """
+
+    X, y = make_classification()
+    y = np.column_stack([y for _ in range(n_outputs_)]).squeeze()
+
+    # loss functions for each output and joined show up as metrics
+    metric_idx = 1 + (n_outputs_ if n_outputs_ > 1 else 0)
+    prefix = "out1_" if n_outputs_ > 1 else ""
+
+    expected_name = metrics
+    if metrics != "binary_accuracy":
+        # Test discovery in VSCode fails if TF prints out _any_
+        # warnings during discovery
+        # (which of course it does if you try to instantiate anything)
```

Comment:
I don't think these comments are necessary?

Reply:
I felt that there is no point in passing a class if it's going to be used as an instance. I wanted to clarify why I did that.

Reply:
I meant the comment on "Test discovery in VSCode fails if TF prints out _any_ warnings during discovery"

Reply:
Maybe I'm not understanding. I put that warning there so that the next person looking at the tests doesn't decide to "improve" them by passing instances to `pytest.mark.parameterize`. Does that make sense? Is it clear enough without the comments?

Reply:
Would ignoring that warning help? https://docs.pytest.org/en/stable/warnings.html#pytest-mark-filterwarnings

Reply:
**TLDR: deleted the comments, keeping it as a class.**

Turns out I'm not even able to reproduce the issue anymore. Maybe last VSCode update fixed it? Not sure, but I'm removing the comment. Funny enough though we still need to have the class instead of the instance because if you pass the same metric instance to multiple outputs, Keras does funky stuff with the naming that breaks this test.